File: python/medium/group_anagram_pairs.py
def group_anagram_pairs(a_list):
  a_dict = {}
  for item in a_list:
    sorted_item = ''.join(sorted(item.strip().lower()))
    if sorted_item in a_dict:
      a_dict[sorted_item].append(item)
    else:
      a_dict[sorted_item] = [item]
  return list(a_dict.values())

print(group_anagram_pairs(['ann', 'nan', 'd', 'pie', 'cort', 'iep']))

File: js/medium/groupAnagramPairs.js
function groupAnagramPairs(arr) {
  var hashTable = {};
  for (var i = 0; i < arr.length; i++) {
    var sortedItem = arr[i].toLowerCase().trim().split('').sort().join('');
    if (hashTable[sortedItem]) {
      hashTable[sortedItem].push(arr[i]);
    } else {
      hashTable[sortedItem] = [arr[i]];
    }
  }
  return Object.values(hashTable);
}
console.log(groupAnagramPairs(['ann', 'nan', 'd', 'pie', 'cort', 'iep']));
